x/liquidity/spec/03_state_transitions.md

```diff
@@ -1,174 +1,235 @@
+- `swapPrice(i)` = (`X` + 2_`EX(i)`)/(`Y` + 2_`EY(i-1)`)
+
+  - condition1) `orderPrice(i)` < `swapPrice(i)` < `orderPrice(i-1)`
+
+- `PoolX(i)` = (`X` - `swapPrice(i)`*`Y`)/2
+
+  - condition2) `PoolX(i)` >= 0
+
+- If both above conditions are met, `swapPrice` is the swap price for this iteration
+
+  - Amount of Y coins matched = `EY(i)`
+
+- If one of these conditions doesn't hold, go to FractionalMatch
+
+- FractionalMatch: SwapPrice is found at an orderPrice
+
+- `swapPrice(i)` = `orderPrice(i)`
+
+- `PoolX(i)` = (`X` - `swapPrice(i)`*`Y`)/2
+
+- Amount of Y coins matched:
+
+  - `EY(i)` ← min[ `EY(i)`, (`EX(i)`+`PoolX(i)`)/`swapPrice(i)` ]
 
 - Find optimized swapPrice
-  - Find `swapPrice(k)` which has the largest amount of Y coins matched
+
+  - Find `swapPrice(k)` that has the largest amount of Y coins matched
+
     - this is our optimized swap price
     - corresponding swap result variables
+
       - `swapPrice(k)`, `EX(k)`, `EY(k)`, `PoolX(k)`
 
-**Calculate matching result**
+### Calculate matching result
 
 - for swap orders from X to Y
 
-  - Iteration : iterate `orderPrice(i)` of swap orders from X to Y (high to low)
+  - Iteration: iterate `orderPrice(i)` of swap orders from X to Y (high to low)
+
     - sort by order price (high to low), sum all order amount with each `orderPrice(i)`
     - if `EX(i)` ≤ `EX(k)`
+
       - `fractionalRatio` = 1
+
     - if `EX(i)` > `EX(k)`
+
       - `fractionalRatio(i)` = (`EX(k)` - `EX(i-1)`) / (`EX(i)` - `EX(i-1)`)
       - break the iteration
-    - matching amount for swap orders with this `orderPrice(i)` :
-      - `matchingAmt` = `offerAmt` \* `fractionalRatio(i)`
+
+    - matching amount for swap orders with this `orderPrice(i)`:
+
+      - `matchingAmt` = `offerAmt` * `fractionalRatio(i)`
 
 - for swap orders from Y to X
-  - Iteration : iterate `orderPrice(i)` of swap orders from Y to X (low to high)
+
+  - Iteration: iterate `orderPrice(i)` of swap orders from Y to X (low to high)
+
     - sort by order price (low to high), sum all order amount with each `orderPrice(i)`
     - if `EY(i)` ≤ `EY(k)`
+
       - `fractionalRatio` = 1
+
     - if `EY(i)` > `EY(k)`
+
       - `fractionalRatio(i)` = (`EY(k)` - `EY(i-1)`) / (`EY(i)` - `EY(i-1)`)
       - break the iteration
-    - matching amount for swap orders with this `orderPrice(i)` :
-      - `matchingAmt` = `offerAmt` \* `fractionalRatio(i)`
 
-**2) Swap Fee Payment**
+    - matching amount for swap orders with this `orderPrice(i)`:
+
+      - `matchingAmt` = `offerAmt` * `fractionalRatio(i)`
+
+### Swap Fee Payment
 
 - Swap fees are calculated after above calculation process
 - Swap fees are proportional to the coins received from matched swap orders
-  - `SwapFee` = `ReceivedMatchedCoin` \* `SwapFeeRate`
+
+  - `SwapFee` = `ReceivedMatchedCoin` * `SwapFeeRate`
+
 - Swap fees are sent to the liquidity pool
 
-**3) Cancel unexecuted swap orders with expired CancelHeight**
+## Cancel unexecuted swap orders with expired CancelHeight
```

Comment:
I didn't understand why these (whatever they are) cases were numbered, can you help me understand?  I think we can just use headings